glean-core/src/lib.rs

```diff
@@ -885,6 +885,10 @@ static FD_LOGGER: OnceCell<fd_logger::FdLogger> = OnceCell::new();
 pub fn glean_enable_logging_to_fd(fd: u64) {
     // SAFETY: TODO.
     unsafe {
+        // If the logger is already set, do not create new logger.
+        if FD_LOGGER.get().is_some() {
+            return;
+        }
```

Comment:
Seems like we should just make use of `OnceCell`'s `get_or_init` for exactly that purpose.

Reply:
`get_or_init` returns the reference to the cell's content, regardless of whether the cell was empty or not.
https://docs.rs/once_cell/latest/once_cell/sync/struct.OnceCell.html#method.get_or_init
So, the remaining operation (`log::set_logger` etc) cannot be written conditional, unless we somehow return the state.

I think the early return is simpler.

Reply:
ah, good to know :)
I'll fix.